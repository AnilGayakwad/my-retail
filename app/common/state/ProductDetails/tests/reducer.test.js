
import { fromJS } from 'immutable';
import ProductDetailsReducer from '../reducer';
import {
  productDetailsRequested,
  productDetailsRequestSucceeded,
  productDetailsRequestFailed,
} from '../actions';

describe('ProductDetails Reducer', () => {
  describe('DEFAULT', () => {
    it('returns the initial state', () => {
      const initialState = ProductDetailsReducer(undefined, {}).toJS();
      expect(initialState).toMatchSnapshot();
    });
  });

  describe('PRODUCT_DETAILS_REQUESTED', () => {
    it('should set isFetching for productId=123', () => {
      const state = fromJS({
        ProductDetails: {
          123: {
            meta: {
              isFetching: false,
            },
          },
        },
      });
      const action = productDetailsRequested('123');
      expect(ProductDetailsReducer(state, action)).toMatchSnapshot();
    });
  });

  describe('PRODUCT_DETAILS_REQUEST_SUCCEEDED', () => {
    beforeEach(() => {
      Date.nowTemp = Date.now;
      Date.now = () => 123123;
    });

    afterEach(() => {
      Date.now = Date.nowTemp;
    });
    it('should wipe errorKey, set isFetching false for productId=123', () => {
      const state = fromJS({
        123: {
          map: { },
          error: {
            errorKey: 'should be gone',
          },
          meta: {
            isFetching: true,
          },
        },
      });
      const action = productDetailsRequestSucceeded({ productDetails: { } }, '123');
      expect(ProductDetailsReducer(state, action)).toMatchSnapshot();
    });
  });

  describe('PRODUCT_DETAILS_REQUEST_FAILED', () => {
    it('should set errorKey to getProductDetailsFailed and isFetching to false for productId=123', () => {
      const state = fromJS({
        123: {
          error: {
            errorKey: '',
          },
          meta: {
            isFetching: true,
          },
        },
      });
      const action = productDetailsRequestFailed(new Error(), '123');
      expect(ProductDetailsReducer(state, action)).toMatchSnapshot();
    });
  });
});
