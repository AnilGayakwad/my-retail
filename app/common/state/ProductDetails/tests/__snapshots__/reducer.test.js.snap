// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDetails Reducer DEFAULT returns the initial state 1`] = `Object {}`;

exports[`ProductDetails Reducer PRODUCT_DETAILS_REQUEST_FAILED should set errorKey to getProductDetailsFailed and isFetching to false for productId=123 1`] = `
Immutable.Map {
  123: Immutable.Map {
    error: Immutable.Map {
      errorKey: "getProductDetailsFailed",
    },
    meta: Immutable.Map {
      isFetching: false,
    },
  },
}
`;

exports[`ProductDetails Reducer PRODUCT_DETAILS_REQUEST_SUCCEEDED should wipe errorKey, set isFetching false for productId=123 1`] = `
Immutable.Map {
  123: Immutable.Map {
    map: Immutable.Map {
      productDetails: Immutable.Map {
      },
    },
    error: Immutable.Map {
      errorKey: "",
    },
    meta: Immutable.Map {
      isFetching: false,
      lastUpdated: 123123,
    },
  },
}
`;

exports[`ProductDetails Reducer PRODUCT_DETAILS_REQUESTED should set isFetching for productId=123 1`] = `
Immutable.Map {
  ProductDetails: Immutable.Map {
    123: Immutable.Map {
      meta: Immutable.Map {
        isFetching: false,
      },
    },
  },
  123: Immutable.Map {
    error: Immutable.Map {
      errorKey: "",
    },
    meta: Immutable.Map {
      isFetching: true,
    },
  },
}
`;
